### ISSUE TITLE
shared library files have incomplete soname (patch)

### ISSUE THREAD

**Issue by brunopostle** [IfcOpenShell/IfcOpenShell#937]
On a unix/linux system shared libraries are expected to be installed using a versioned [soname](https://en.wikipedia.org/wiki/Soname), with the unversioned libXXX.so name existing only as a symlink. The attached patch to CMakeLists.txt fixes this for the v0.6.0 branch. The result looks something like this: 

`lrwxrwxrwx. 1 root root       15 Jul 29 23:53 /usr/lib64/libIfcGeom.so -> libIfcGeom.so.0`
`lrwxrwxrwx. 1 root root       19 Jul 29 23:53 /usr/lib64/libIfcGeom.so.0 -> libIfcGeom.so.0.0.0`
`-rwxr-xr-x. 1 root root  4581336 Jul 29 23:53 /usr/lib64/libIfcGeom.so.0.0.0`
`lrwxrwxrwx. 1 root root       16 Jul 29 23:53 /usr/lib64/libIfcParse.so -> libIfcParse.so.0`
`lrwxrwxrwx. 1 root root       20 Jul 29 23:53 /usr/lib64/libIfcParse.so.0 -> libIfcParse.so.0.0.0`
`-rwxr-xr-x. 1 root root 32844392 Jul 29 23:53 /usr/lib64/libIfcParse.so.0.0.0`
`lrwxrwxrwx. 1 root root       19 Jul 30 00:14 /usr/lib64/libSerializers.so -> libSerializers.so.0`
`lrwxrwxrwx. 1 root root       23 Jul 30 00:14 /usr/lib64/libSerializers.so.0 -> libSerializers.so.0.0.0`
`-rwxr-xr-x. 1 root root  1001416 Jul 30 00:14 /usr/lib64/libSerializers.so.0.0.0`

[ifcopenshell-soname.patch](https://github.com/IfcOpenShell/IfcOpenShell/files/4997935/ifcopenshell-soname.patch.txt)

**Comment by aothms** [comment]
Thanks, I think that makes sense. Any reason you didn't make this a PR? Should the version number be 0.6 or something?

**Comment by brunopostle** [comment]
I'd forgotten how to do a pull request, I'll do one.

The soname version numbers tend not to follow release numbers: the first number increments whenever there is a binary API change, so it may change between IfcOpenShell 0.6 and 0.7 for example. The second and third numbers are for API compatible major/minor releases. So I started at 0.0.0.

Note that I've only tested this on Linux, it shouldn't affect the DLL name on Windows where the number is stored internally.
